TUTORIAL.md

```diff
@@ -313,6 +268,78 @@ in
 module.id module.pi
 ```
 
+## Typesystem
+
+### Function types
+
+```
+<type> -> <type>
+```
+
+Function types are written using the `->` operator which is right associative. This means that the type of `(+)` which is usually written as `Int -> Int -> Int` is parsed as `Int -> (Int -> Int)` (A function taking one argument of `Int` and returning a function of `Int -> Int`).
+
+### Record type
+
+```
+{ (<identifier> <identifier>* = <type>,)* (<identifier> : <type>,)* }
+{ pi : Float, sin : Float -> Float }
+```
+
+Records are gluon's main way of creating associating related data and they should look quite familiar if you are familiar with dynamic languages such as javascript. Looks can be deceiving however as gluon's records are more similiar to a struct in Rust or C as the order of the fields are significant and fields can't be added, removed or modified.
```

Comment:
That  `{ x : Int, y : String } != { y : String, x : Int }`. It might be a bit cumbersome but it lets the interpreter access fields just by offset instead of by an hashmap. I can clarify it a bit.